core/src/com/po/fuck/model/GameObject.java

```diff
@@ -15,6 +16,10 @@ public class GameObject implements Drawable, Collidable, Destructable {
         this.position = position;
     }
 
+    public Sprite getSprite(){
+        return new Sprite(this.sprite);
+    }
+
     @Override
     public void draw(CenterDrawer drawer) {
```

Comment:
It is not a `model`'s logic

Reply:
`draw` method was left to have a possibility of easy classification between `Drawable` and `non-Drawable` objects. If it were better, we could make an `Interface Drawable` that does not have any methods defined.

Reply:
I think it should have method `getPosition() -> Vector2` and nothing else

Reply:
However, I would ask how we can implement `getPosition` for classes like `HealthBar`, where `getPosition` relies on the position of `Entity` AND size of its `Sprite`, because currently I have no ideas.

Reply:
I am thinking of adding `Drawer` for each like in the similar way it was added in `Manager`.

Reply:
> However, I would ask how we can implement `getPosition` for classes like `HealthBar`, where `getPosition` relies on the position of `Entity` AND size of its `Sprite`, because currently I have no ideas.

I have already mentioned this before, we need to move size out of a sprite, and do not relate on the sprite anymore

Reply:
How can we move out the size of the sprite, if it could be changed with the Sprite itself? Also, when object collides we need size as well...

Reply:
> I am thinking of adding `Drawer` for each like in the similar way it was added in `Manager`.

I do not get it, what do you want to add ? Manager do not have any `forEach`.

Reply:
I want to add in `Renderer` field `Map<Class<T>, drawingFunction(position, Sprite, CentralDrawer)>` and initialize it during the creation of the `Renderer` class

Reply:
But still you should move drawing functionality out of the model. You can make `Map<Class<?>, func(Object object, CentralDrawer)>`, and initialize it statically for each class.
We can assume, that the most model have some simple drawing logic like `drawer.draw(sprite, position)`, and create some function factory passing `sprite` to it.